solr/core/src/java/org/apache/solr/storage/AccessDirectory.java

```diff
@@ -578,14 +580,20 @@ private static ByteBufferGuard.BufferCleaner unmapHack() {
     }
   }
 
-  static final class LazyLoadInput extends IndexInput implements RandomAccessInput {
+  public interface LazyLoad {
```

Comment:
Is this interface necessary right now? It looks like we don't reference anything by this interface, and only have one implementation of it. Is this something that's more forward looking, for when we are upstreaming this in the future?

Reply:
Thanks! yeah; this is similar to the above -- this interface is used in testing by external code. I added a comment to this effect in 6fb2dab210625079cd70e9b60d8252a3968b9161